File: frontend/src/types/invoice-payment.ts
export interface InvoicePayment {
  id: string;
  invoice_id: string;
  amount: string;
  payment_date: string;
  payment_method: string | null;
  reference: string | null;
  bank_transaction_id: string | null;
  journal_entry_id: string | null;
  created_at: string;
}

export interface RecordPaymentData {
  amount: number;
  payment_date: string;
  bank_account_id: string;
  /** Required only when the bank account's currency differs from the invoice's. */
  actual_base_amount?: number;
  payment_method?: string;
  reference?: string;
}

File: backend/crates/konto-db/src/entities/invoice_payment.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "invoice_payments")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: String,
    pub invoice_id: String,
    #[sea_orm(column_type = "Decimal(Some((15, 2)))")]
    pub amount: rust_decimal::Decimal,
    pub payment_date: chrono::NaiveDate,
    pub payment_method: Option<String>,
    pub reference: Option<String>,
    pub bank_transaction_id: Option<String>,
    pub journal_entry_id: Option<String>,
    pub created_at: chrono::NaiveDateTime,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "super::invoice::Entity",
        from = "Column::InvoiceId",
        to = "super::invoice::Column::Id"
    )]
    Invoice,
}

impl Related<super::invoice::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::Invoice.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
